security/src/main/java/io/airlift/security/client/SpnegoAuthentication.java

```diff
@@ -0,0 +1,119 @@
+package io.airlift.security.client;
+
+import io.airlift.log.Logger;
+import io.airlift.security.AuthScheme;
+import io.airlift.security.exception.AuthenticationException;
+import io.airlift.security.utils.KerberosUtil;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.Attributes;
+import org.eclipse.jetty.util.B64Code;
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSException;
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.LoginException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class SpnegoAuthentication
+        extends JettyAuthentication
+{
+    private static final String NEGOTIATE = HttpHeader.NEGOTIATE.asString();
+    private static final Logger log = Logger.get(SpnegoAuthentication.class);
+    private final String krb5Conf;
+    private final String serviceName;
+    private GSSContext gssContext = null;
```

Comment:
Good catch! GSSContext should not be cached. It is only cached on the server-side.

BTW, does JettyHttpClient allow concurrent requests to the same host?